src/webpack-angular-types-plugin/plugin.ts

```diff
@@ -12,40 +12,51 @@ import { getPrototypeComponentIDCodeBlock } from "./templating/component-global-
 import { generateClassInformation } from "./type-extraction/type-extraction";
 
 export class WebpackAngularTypesPlugin {
+    // A queue for modules, that should be processed by the plugin in the next seal-hook
+    private moduleQueue: Module[] = [];
+
+    // A dummy tsProject that is used to gather the glob-paths from the "include"
+    // field of the tsconfig file (therefore all dependency resolution etc is skipped)
+    // TODO maybe this can be replaced with a super-fast glob implementation
+    private tsProject = new Project({
+        tsConfigFilePath: DEFAULT_TS_CONFIG_PATH,
+        skipLoadingLibFiles: true,
+        skipFileDependencyResolution: true,
+    });
+
     apply(compiler: Compiler) {
-        const tsProject = new Project({
-            tsConfigFilePath: DEFAULT_TS_CONFIG_PATH,
-            skipLoadingLibFiles: true,
-            skipFileDependencyResolution: true,
-        });
         compiler.hooks.compilation.tap(PLUGIN_NAME, (compilation) => {
             compilation.dependencyTemplates.set(
                 CodeDocDependency,
                 new CodeDocDependencyTemplate()
             );
+            compilation.hooks.buildModule.tap(PLUGIN_NAME, (module) => {
+                this.moduleQueue.push(module);
+            });
             compilation.hooks.seal.tap(PLUGIN_NAME, () => {
-                const modulesToProcess = this.collectModulesToProcess(
-                    compilation.modules,
-                    tsProject
-                );
-                // TODO check if this approach is actually faster
                 const smallTsProject = new Project({
                     tsConfigFilePath: DEFAULT_TS_CONFIG_PATH,
+                    skipLoadingLibFiles: true,
                     skipAddingFilesFromTsConfig: true,
                     skipFileDependencyResolution: true,
                 });
-                for (const moduleToProcess of modulesToProcess) {
-                    smallTsProject.addSourceFileAtPath(moduleToProcess[0]);
+
+                const modulesToProcess = this.moduleQueue
+                    .map((module) => this.getModuleInfoIfProcessable(module))
+                    .filter((module): module is ModuleInformation => !!module);
```

Comment:
For performance, it would be better to first filter, then map. Otherwise, it iterates over all items of the list twice instead of mapping just the filtered list